gh-badges/lib/index.js

```diff
@@ -17,6 +19,56 @@ class BadgeFactory {
     }
   }
 
+  _validate(format) {
+    if (!('text' in format)) {
+      throw new ValidationError('Field `text` is required')
+    }
+
+    if (
+      !Array.isArray(format.text) ||
+      format.text.length !== 2 ||
+      typeof format.text[0] !== 'string' ||
+      typeof format.text[1] !== 'string'
+    ) {
+      throw new ValidationError('Field `text` must be an array of 2 strings')
+    }
+
+    const stringFields = [
+      'labelColor',
+      'color',
+      'colorA',
+      'colorscheme',
+      'colorB',
+    ]
+    stringFields.forEach(function(field) {
+      if (field in format && typeof format[field] !== 'string') {
+        throw new ValidationError(`Field \`${field}\` must be of type string`)
+      }
+    })
+
+    const formatValues = ['svg', 'json']
```

Comment:
We never came to an agreement on this. The discussion in https://github.com/badges/shields/pull/3586#discussion_r295802787 still applies here.

Reply:
I don't have any strong feelings on this one myself. I suppose that either approach would be a breaking API change. 

I'd probably vote for the behavior you've implemented here (throwing a validation error on raster format). Obviously the lib doesn't support raster today, and AFAICT it doesn't seem like there's been any requests for it.

However, I'd be okay with adding raster support in the lib if we intend to keep raster support in the CLI, and especially if the process of fixing/upgrading the CLI support (refs https://github.com/badges/shields/issues/1788) could be aided by doing so.

Reply:
I agree for v3 we should either add raster to the library or remove it from the CLI so the two are consistent.

re: breaking changes, adding validation at all is a breaking change as explained in https://github.com/badges/shields/pull/3507#issue-283411297 so I'm not really worried about making a non-backwards-compatible change here. Lets just pick the way we want it to be moving forwards.

Basically what I want to do is work on each of the things listed in #3587 as a seperate PRs, merge them all to `v3-dev` and then when all that is done, merge that to master. That way we can still do a patch release off master in the meantime if we need to but we don't have to load all the v3 changes into one PR. I appreciate that for various reasons, bullet point one on that list has dragged out over about 3 months, so we've kinda lost sight of the high-level stuff a bit.